### ISSUE TITLE
LDTK can open, but can not save this file.

### ISSUE THREAD

**Issue by XTremEive** [deepnight/ldtk#751]
Hi,

So I have the following level file:
https://cdn.discordapp.com/attachments/430653956415422495/1046981128210354286/Woods_1_0.ldtk

It opens just fine, but even if you DO NOT CHANGE anything and save it, LDTK will output a corrupted LDTK file which can not be opened anymore.

Here is the error report you get AFTER trying to open the, corrupted, saved version:

Regards.

Stack:
```
LDtk version: 1.1.3-64bits
Cannot read properties of null (reading 'externalLevels')
TypeError
TypeError: Cannot read properties of null (reading 'externalLevels')
    at Object.cb (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:214580:8)
    at ui_modal_Progress.update (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:220409:8)
    at Function.dn_Process._doMainUpdate (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:223:4)
    at Function.dn_Process.updateAll (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:734:14)
    at Boot.update (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:4571:14)
    at Boot.mainLoop (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:4533:8)
    at hxd_System.browserLoop (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:138601:14)
```
Processes:
```
#0 App
 |--- #2 page.Home
 |--- #4 page.Home
#1 ui.modal.Progress
#3 ui.modal.dialog.LogPrint
#5 ui.modal.dialog.LogPrint
```

Log:
```


[BOOT]         App started
[BOOT]         Version: 1.1.3-64bits (build 458364)
[BOOT]         ExePath: C:/Users/user/AppData/Local/Programs/ldtk
[BOOT]         Assets: C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets
[BOOT]         ExtraFiles: C:/Users/user/AppData/Local/Programs/ldtk/extraFiles
[BOOT]         CWD: D:\StudioName\Rpg01\Rpg01\Assets\Definitions\Rooms\Woods_1_0\
[BOOT]         Display: 1920x1080
[BOOT]         Args: soloValues=[D:\StudioName\Rpg01\Rpg01\Assets\Definitions\Rooms\Woods_1_0\Woods_1_0.ldtk], args=[]
[FILE]         Loading settings from C:/Users/user/AppData/Roaming/LDtk/settings...
[BOOT]         AppZoomFactor: 1
[BOOT]         Calling appReady...
[UPDATE]       Looking for update
[BOOT]         Start args: path=D:\StudioName\Rpg01\Rpg01\Assets\Definitions\Rooms\Woods_1_0\Woods_1_0.ldtk levelIndex=null
[BOOT]         Loading project from args (D:\StudioName\Rpg01\Rpg01\Assets\Definitions\Rooms\Woods_1_0\Woods_1_0.ldtk)...
[GENERAL]      Progress created.
[GENERAL]      Progress started: : "Woods_1_0.ldtk: Project...", 5 operation(s)
[LOAD]         Loading project Woods_1_0.ldtk...
[ERROR]        SyntaxError: Unexpected token _ in JSON at position 185008
[ERROR]        Failed to load project: D:\StudioName\Rpg01\Rpg01\Assets\Definitions\Rooms\Woods_1_0\Woods_1_0.ldtk levelIdx=null
[GENERAL]      Page started: page.Home()
[FILE]         Loading page template: home from C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/home.html
[DEBUG]        samplesDir=C:/Users/user/AppData/Local/Programs/ldtk/extraFiles/samples
[FILE]         Loading HTML template logPrint
[ERROR]        TypeError: Cannot read properties of null (reading 'jsonVersion')
[ERROR]        Failed to load project: D:\StudioName\Rpg01\Rpg01\Assets\Definitions\Rooms\Woods_1_0\Woods_1_0.ldtk levelIdx=null
[GENERAL]      Page started: page.Home()
[FILE]         Loading page template: home from C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/home.html
[DEBUG]        samplesDir=C:/Users/user/AppData/Local/Programs/ldtk/extraFiles/samples
[FILE]         Reading HTML template logPrint from cache
[GENERAL]      Page started: page.CrashReport()
[FILE]         Loading page template: crashReport from C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/tpl/pages/crashReport.html
[ERROR]        Cannot read properties of null (reading 'externalLevels') (TypeError)
[ERROR]        TypeError: Cannot read properties of null (reading 'externalLevels')
    at Object.cb (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:214580:8)
    at ui_modal_Progress.update (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:220409:8)
    at Function.dn_Process._doMainUpdate (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:223:4)
    at Function.dn_Process.updateAll (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:734:14)
    at Boot.update (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:4571:14)
    at Boot.mainLoop (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:4533:8)
    at hxd_System.browserLoop (file:///C:/Users/user/AppData/Local/Programs/ldtk/resources/app.asar/assets/js/renderer.js:138601:14)

[GENERAL]      
#0 App
 |--- #2 page.Home
 |--- #4 page.Home
 |--- #6 page.CrashReport
#1 ui.modal.Progress
#3 ui.modal.dialog.LogPrint
#5 ui.modal.dialog.LogPrint

```

**Comment by XTremEive** [comment]
Hack:
- Using the rename button in LDTK to change the name of the project, and then change it back to its original name fixes the issue.

The file can then be saved again.

However, if you don't have a backup before the corruption happens you're dooooooomed!!! ... I guess. 

Best regards.

**Comment by deepnight** [comment]
Did you rename the file manually outside of LDtk?

**Comment by XTremEive** [comment]
> Did you rename the file manually outside of LDtk?
Yes I did.
Yes I shouldn't. (I didn't know the warning was there.)

**Comment by deepnight** [comment]
> > Did you rename the file manually outside of LDtk?
> Yes I did.
> Yes I shouldn't. (I didn't know the warning was there.)

Thanks, I need to find a way to detect this to notify the user properly 🤔

**Comment by Cammin** [comment]
I setup my importer so users can't rename LDtk files in unity. Good thing I did :)

**Comment by deepnight** [comment]
Fixed in 1.2.0
